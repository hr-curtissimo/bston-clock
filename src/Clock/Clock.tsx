import * as React from 'react';

interface IClockProps {
  time: number;
  timezone: { name: string; offset: number };
}

function formatDate(d: Date) {
  return `${d.getHours()}:${d.getMinutes()}:${d.getSeconds()}`;
}

const Clock = (props: IClockProps) => {
  const t = new Date(props.time - (props.timezone.offset) * 1000 * 60)

  return <time>{formatDate(t)} {props.timezone.name}</time>
};

export default Clock;
