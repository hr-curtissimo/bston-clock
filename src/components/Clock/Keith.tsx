import * as React from 'react';

export const Keith = () => (
  <div>KEITH!</div>
);
