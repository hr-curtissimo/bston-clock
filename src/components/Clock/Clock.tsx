import * as React from 'react';
import { connect } from 'react-redux';
import { IState } from '../../store/reducers';
import { Keith as KC } from './Keith';

interface IClockProps {
  time: number;
  timezone: { name: string; offset: number };
}

function formatDate(d: Date) {
  return `${d.getHours()}:${d.getMinutes()}:${d.getSeconds()}`;
}

export const Clock = (props: IClockProps) => {
  const t = new Date(props.time - (props.timezone.offset) * 1000 * 60)

  return (
    <time>{formatDate(t)} <KC /> {props.timezone.name}</time>
  )
};

const mapStateToProps = (state: IState) => {
  return {
    time: state.time
  };
};

export default connect(mapStateToProps)(Clock);
