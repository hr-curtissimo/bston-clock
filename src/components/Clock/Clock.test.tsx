import { mount, render, shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Clock } from './Clock';

import { Keith } from './Keith';

test('shallow: Clock renders time with timezone', () => {
  // Arrange
  const time = 123;
  const timezone = { name: 'BOB', offset: -3.14 };

  // Act
  const component = shallow(<Clock time={time} timezone={timezone} />);

  // Assert
  expect(component).not.toBeNull()
  expect(component.find(Keith)).not.toBeNull();
  expect(component.html()).toContain('BOB');
  expect(component.html()).toContain('19:3:8');
});

test('render: Clock renders time with timezone', () => {
  // Arrange
  const time = 123;
  const timezone = { name: 'BOB', offset: -3.14 };

  // Act
  const component = render(<Clock time={time} timezone={timezone} />);

  // Assert
  expect(component).not.toBeNull()
  expect(component.text()).toContain('19:3:8 KEITH! BOB');
});

test('mount: Clock renders time with timezone', () => {
  // Arrange
  const time = 123;
  const timezone = { name: 'BOB', offset: -3.14 };

  // Act
  const component = mount(<Clock time={time} timezone={timezone} />);

  // Assert
  expect(component).not.toBeNull()
  expect(component.text()).toContain('19:3:8 KEITH! BOB');
});

test('mount: Clock renders time with timezone', () => {
  // Arrange
  let timeout = false;
  const handle = setTimeout(() => timeout = true, 1000);

  const time = 123;
  const timezone = { name: 'BOB', offset: -3.14 };

  // Act
  const component = mount(<Clock time={time} timezone={timezone} />);

  // Assert
  return new Promise((good, bad) => {
    setTimeout(() => {
      clearTimeout(handle);
      if (timeout) { bad('exceeded timeout'); }
      expect(component).not.toBeNull()
      expect(component.text()).toContain('19:3:8 KEITH! BOB');
      good();
    }, 0);
  });
});

test('renders consistently', () => {
  // ARRANGE
  const time = 123;
  const timezone = { name: 'BOB', offset: -3.14 };

  // ACTu
  const tree = renderer
    .create(<Clock time={time} timezone={timezone}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
