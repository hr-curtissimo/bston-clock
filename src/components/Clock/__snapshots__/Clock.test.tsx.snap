// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders consistently 1`] = `
<time>
  19:3:8
   
  <div>
    KEITH!
  </div>
   
  BOB
</time>
`;
